File: Model/Brand.cs
namespace sep3.Model;

public class Brand
{
    private long Id { get; set; }
    private string Name { get; set; }
    private List<Product> Products;

    public Brand()
    {
    }

    public Brand(List<Product> products, long id, string name)
    {
        Products = products;
        Id = id;
        Name = name;
    }
}

File: warehouse/src/main/java/sep3/warehouse/entities/Brand.java
package sep3.warehouse.entities;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Table(name = "brand")
public class Brand {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, length = 50)
    private String name;

    @OneToMany(mappedBy = "brand", fetch = FetchType.LAZY)
    private List<Product> products;
}

